tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -53,6 +55,27 @@ public TraceContext(IDatadogTracer tracer, TraceTagCollection tags = null)
             get => _samplingPriority;
         }
 
+        public bool IastInitialized
+        {
+            get => _iastRequestContext != null;
+        }
+
+        /// <summary>
+        /// Gets the iast context.
+        /// </summary>
+        internal IastRequestContext IastRequestContext
+        {
+            get
+            {
+                if (_iastRequestContext == null && Iast.Iast.Instance.Settings.Enabled)
+                {
+                    _iastRequestContext = new();
+                }
```

Comment:
There is a race condition here where multiple threads initialize `_iastRequestContext` and they end up using different `IastRequestContext` instances. You may want to use a lock or maybe something like `Interlocked.CompareExchange()`.